pysr/ssr.py

```diff
@@ -0,0 +1,171 @@
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_times, n_features).
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`.
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0t_1", "x1t_2", etc.
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        if y is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `y` - this parameter is being ignored"
+            )
+        if y_units is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `y_units` - this parameter is being ignored"
+            )
+        if Xresampled is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `Xresampled` - this parameter is being ignored"
+            )
+
+        if self.recursive_history_length <= 0:
+            raise ValueError(
+                "The `recursive_history_length` parameter must be greater than 0 (otherwise it's not recursion)."
+            )
+        if len(X.shape) > 2:
+            raise ValueError(
+                "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+            )
+        elif len(X) < 2:
+            raise ValueError(
+                "Recursive symbolic regression requires at least 2 datapoints; if you tried to pass a 1D array, use array.reshape(-1, 1)"
+            )
+        if len(X) <= self.recursive_history_length + 1:
+            raise ValueError(
+                f"Recursive symbolic regression with a history length of {self.recursive_history_length} requires at least {self.recursive_history_length + 2} datapoints."
+            )
+        if isinstance(weights, np.ndarray) and len(weights) != len(X):
+            raise ValueError("The length of `weights` must have shape (n_times,).")
+        y = X.copy()
+        temp = X.copy()[0]
```

Comment:
More descriptive name of this? I'm not sure what this is otherwise. Also it would be preferable to copy only the 0th element, rather than taking the 0th element of a copy.